store/src/java/com/zimbra/cs/mailbox/MailboxManager.java

```diff
@@ -756,7 +683,7 @@ public int getMailboxCount() {
     public int lookupMailboxId(String accountId) {
         Integer v;
         synchronized (this) {
-            v = mailboxIds.get(accountId);
+            v = cacheManager.getMailboxKey(accountId);
```

Comment:
I know it, just left the code as it was , it can be seen in `feature/ha` branch https://github.com/Zimbra/zm-mailbox/blob/62f85f467f4b4e45b71db2cebef2e6708e486942/store/src/java/com/zimbra/cs/mailbox/MailboxManager.java#L759 https://github.com/Zimbra/zm-mailbox/blob/62f85f467f4b4e45b71db2cebef2e6708e486942/store/src/java/com/zimbra/cs/mailbox/MailboxManager.java#L834

I thought in not changing that to prevent changing the behavior. 
What should we do? create 2 implementations of `CacheManager#getMailboxKey` , in one of them change accountId to lower case and the other implementation case sansitive? or just one case sensitive implementation and remove `toLowerCase` in method calls ?

Reply:
@israel-arteaga I believe that any place where `toLowerCase` was not being called is a bug. We should just move the `toLowerCase` call into `CacheManager#getMailboxKey` to ensure that it is always done.